src/use-swr-pages.tsx

```diff
@@ -149,29 +149,32 @@ export function useSWRPages<OffsetType = any, Data = any, Error = any>(
         pageSWRs[id].error !== swr.error ||
         pageSWRs[id].revalidate !== swr.revalidate
       ) {
-        setPageSWRs(swrs => {
-          const _swrs = [...swrs]
-          _swrs[id] = {
-            data: swr.data,
-            error: swr.error,
-            revalidate: swr.revalidate,
-            isValidating: swr.isValidating,
-            mutate: swr.mutate
+        // hoist side effects: setPageSWRs and setPageOffsets -- https://reactjs.org/blog/2020/02/26/react-v16.13.0.html#warnings-for-some-updates-during-render
+        setTimeout(() => {
```

Comment:
Another way to hoist is `useEffect()`s, possibly with dependency arrays to take the place of the `if` clauses below.